gradle/dist.gradle

```diff
@@ -1,53 +1,87 @@
+        [name: 'gradle-wrapper'],
+        [name: 'fo'],
+        [name: 'eclipsehelp'],
+        [name: 'eclipsehelp'],
+        [name: 'htmlhelp'],
+        [name: 'htmlhelp'],
+        [name: 'xep'],
+        [name: 'axf'],
+        [name: 'dost'],
+        [name: 'error_prone_annotations'],
+        [name: null, title: 'DITA 1.1 DTDs and Schemas', license: ['oasis-dita11.txt']],
+        [name: null, title: 'DITA 1.2 DTDs and Schemas', license: ['oasis-dita12.txt']]
+]
 
 task generateNotices() {
     doLast {
-        println("jars")
         file(distTempDir).traverse(type: FILES, nameFilter: ~/.*\.jar$/) {
-            if (it.getParentFile().getName() != "lib") {
-                // println("skip ${it}")
-            } else {
-                switch(it.getName()) {
-                    case ~/^batik-.*$/:
-                    case ~/^commons-.*$/:
-                    case ~/^avalon-.*$/:
-                    case ~/^fop-.*$/:
-                    case ~/^ant-.*$/:
-                    case ~/^ant\..*$/:
-                    case ~/^xml-apis-.*$/:
-                    case ~/^xercesImpl-.*$/:
-                    case ~/^xmlgraphics-commons-.*$/:
-                    case ~/^xml-resolver-.*$/:
-                    case ~/^serializer-.*$/:
-                    case ~/^fontbox-.*$/:
-                    case ~/^jackson-.*$/:
-                    case ~/^guava-.*$/:
-                        copyLicense("apache2.txt", it)
-                        break;
-                    case ~/^slf4j-.*$/:
-                    case ~/^jcl-over-slf4j-.*$/:
-                        copyLicense("mit-slf4j.txt", it)
-                        break;
-                    case ~/^Saxon-.*$/:
-                        copyLicense("mpl2.txt", it)
-                        break;
-                    case ~/^logback-.*$/:
-                        copyLicense("epl-1.0.txt", it)
-                        break;
-                    case ~/^icu4j-.*$/:
-                        copyLicense("icu.txt", it)
-                        break;
-                    case ~/^jing-.*$/:
-                        copyLicense("bsd-jing.txt", it)
-                        break;
-                    default:
-                        println "??? ${it}"
+            def name = it.getName()
+            def key = licenses.find { it.name != null && name.startsWith(it.name) }
+            if (key == null) {
+                throw new GradleScriptException("License for ${name} not found", new RuntimeException());
+            }
+        }
+
+        file("${distTempDir}/NOTICES.txt").withWriter('UTF-8') { out ->
+            out.writeLine("""DITA Open Toolkit: Third Party Notices
+
+This Notices file contains certain notices and important information that the
+DITA Open Toolkit Project licensors and contributors (the “Project”) are
+required to provide to you with respect to certain third party components
+included in the DITA Open Toolkit (the “Program”).
```

Comment:
Drop the article per project preferences:

> included in ~the~ DITA Open Toolkit (the “Program”)

Reply:
> @jelovirt I checked out your "feature/notices" branch, how do I generate the "NOTICES.txt"? I invoked the "gradlew" and built the project but the notices file did not appear.

You should run `./gradlew dist`